controllers/openstackdataplanerole_controller.go

```diff
@@ -275,7 +271,7 @@ func (r *OpenStackDataPlaneRoleReconciler) Reconcile(ctx context.Context, req ct
 
 // ReconcileBaremetalSet Reconcile OpenStackBaremetalSet
 func (r *OpenStackDataPlaneRoleReconciler) ReconcileBaremetalSet(ctx context.Context, instance *dataplanev1beta1.OpenStackDataPlaneRole, helper *helper.Helper,
-) (ctrl.Result, error) {
+) (*ctrl.Result, error) {
```

Comment:
If we think moving ahead with deployment and then retrying (for failed  AnsibleExecutionJobs is fine) till baremetalset is ready, we can remove ctrl.Result return value from this function.

Reply:
When I fix the previous comment, this will return &ctrl.Result{} when the baremetalset is not ready, which will cause the main reconcile to stop and return as well.